gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/GradleCacheExplorer.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions.intellij;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GradleCacheExplorer {
+
+    private static final Logger log = LoggerFactory.getLogger(GradleCacheExplorer.class);
+    private static final String GRADLE_CACHE_PATH = System.getProperty("user.home") + "/.gradle/caches/modules-2/";
+    private static final Cache<String, Set<String>> cache =
+            Caffeine.newBuilder().expireAfterWrite(10, TimeUnit.MINUTES).build();
+
+    private final List<String> projectUrls;
+
+    public GradleCacheExplorer(List<String> projectUrls) {
+        this.projectUrls = projectUrls;
+    }
+
+    public final Set<String> getCompletions(DependencyGroup input) {
+        String parsedInput = String.join(".", input.parts());
+        ProgressIndicator indicator = ProgressManager.getInstance().getProgressIndicator();
+        if (indicator == null) {
+            return new HashSet<>();
+        }
+
+        try {
+            Future<Set<String>> future =
+                    ApplicationManager.getApplication().executeOnPooledThread(extractStrings(indicator));
+            Set<String> results =
+                    com.intellij.openapi.application.ex.ApplicationUtil.runWithCheckCanceled(future::get, indicator);
+
+            return results.stream()
```

Comment:
why don't we short-circuit here and do:

```java
if (parsedInput.isEmpty()) {
  return results;
}
```

I believe they are equivalent, but you're not just constructing the set over again